r/tests/testthat/test-schema.R

```diff
@@ -315,5 +315,14 @@ test_that("schema extraction", {
 
   adq <- as_adq(example_data)
   expect_equal(schema(adq), adq$.data$schema)
+})
 
+test_that("schema print truncation", {
+  tbl <- arrow_table(example_data)
+  out <- print_schema_fields(schema(tbl), truncate = TRUE, max_fields = 1)
+  expect_output(
+    cat(out),
+    "int: int32\n...\n6 more columns\nUse `schema()` to see entire schema",
+    fixed = TRUE
+  )
```

Comment:
I assume we already cover the `n_fields < max_fields` case in other tests, yeah?

What happens if `max_fields` is 0 or negative? We might not need to specifically catch these in code, since if I'm reading correctly no end user would ever be able to control that, right? But maybe a comment would be good?

Reply:
Thanks Jon; yeah, that case is covered elsewhere, and I've added in an assertion that `max_fields > 0` and a corresponding test.